internal/kustomize/dependencies_test.go

```diff
@@ -0,0 +1,178 @@
+package kustomize
+
+import (
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/windmilleng/tilt/internal/testutils/tempdir"
+)
+
+func TestNoFile(t *testing.T) {
+	f := newKustomizeFixture(t)
+
+	f.assertErrorContains("kustomization.yaml: no such file or directory")
+}
+
+func TestEmpty(t *testing.T) {
+	f := newKustomizeFixture(t)
+	kustomizeFile := ""
+	f.writeRootKustomize(kustomizeFile)
+
+	expected := []string{"kustomization.yaml"}
+
+	f.assertDeps(expected)
+}
+
+func TestSimple(t *testing.T) {
+	f := newKustomizeFixture(t)
+	kustomizeFile := `# Example configuration for the webserver
+# at https://github.com/monopole/hello
+commonLabels:
+  app: my-hello
+
+resources:
+  - deployment.yaml
+  - service.yaml
+  - configMap.yaml`
+	f.writeRootKustomize(kustomizeFile)
+
+	expected := []string{"kustomization.yaml", "deployment.yaml", "service.yaml", "configMap.yaml"}
+	f.assertDeps(expected)
+}
+
+func TestComplex(t *testing.T) {
+	f := newKustomizeFixture(t)
+	kustomizeFile := `
+# declare ConfigMap as a resource
+resources:
+- configmap.yaml
+
+# declare ConfigMap from a ConfigMapGenerator
+configMapGenerator:
+- name: a-configmap
+  files:
+    - configs/configfile
+    - configs/another_configfile
+
+patchesJson6902:
+  - target:
+      group: extensions
+      version: v1beta1
+      kind: Ingress
+      name: my-ingress
+    path: ingress_patch.yaml`
+	f.writeRootKustomize(kustomizeFile)
+
+	expected := []string{"kustomization.yaml", "configmap.yaml", "ingress_patch.yaml", "configs/configfile", "configs/another_configfile"}
+	f.assertDeps(expected)
+}
+
+func TestRecursive(t *testing.T) {
+	f := newKustomizeFixture(t)
+
+	kustomize := `bases:
+- ./dev
+- ./staging
+- ./production
+
+namePrefix: cluster-a-`
+
+	f.writeRootKustomize(kustomize)
+
+	base := `resources:
+- pod.yaml`
+	f.writeBaseKustomize("base", base)
+	basePod := `apiVersion: v1
+  kind: Pod
+  metadata:
+    name: myapp-pod
+    labels:
+      app: myapp
+  spec:
+    containers:
+    - name: nginx
+      image: nginx:1.7.9`
+	f.writeBaseFile("base", "pod.yaml", basePod)
+
+	dev := `bases:
+- ./../base
+namePrefix: dev-`
+	f.writeBaseKustomize("dev", dev)
+
+	staging := `bases:
+- ./../base
+namePrefix: stag-`
+	f.writeBaseKustomize("staging", staging)
+
+	production := `bases:
+- ./../base
+namePrefix: prod-`
+	f.writeBaseKustomize("production", production)
+
+	expected := []string{
+		"base/kustomization.yaml",
+		"base/pod.yaml",
+		"dev/kustomization.yaml",
+		"staging/kustomization.yaml",
+		"production/kustomization.yaml",
+		"kustomization.yaml",
+	}
+	f.assertDeps(expected)
+}
+
+type kustomizeFixture struct {
+	t       *testing.T
+	tempdir *tempdir.TempDirFixture
```

Comment:
we usually embed TempDirFixture directly, so you can just do f.T() and f.WriteDir() out of the box

https://golang.org/doc/effective_go.html#embedding